src/components/ProposalRow.tsx

```diff
@@ -0,0 +1,58 @@
+import { GenericRow } from "@/components/atoms/GenericRow";
+import { RowType } from "@/components/atoms/types";
+import { ImageWithPlaceholder } from "@/components/atoms/images/ImageWithPlaceholder";
+import { formatUrl } from "@/utils/helper.util";
+import { Box } from "@ledgerhq/react-ui";
+import { useCallback, useMemo } from "react";
+import { useNavigate } from "@tanstack/react-router";
+import { ProposalTypes } from "@walletconnect/types";
+import useAnalytics from "@/hooks/useAnalytics";
+import { useProposal } from "@/hooks/useProposal/useProposal";
+import VerificationLabel from "./verification/VerificationLabel";
+
+const ProposalRow = ({ proposal }: { proposal: ProposalTypes.Struct }) => {
+  const navigate = useNavigate({ from: "/" });
+  const analytics = useAnalytics();
+
+  const { getValidation } = useProposal(proposal);
+  const verificationStatus = useMemo(() => getValidation(), [getValidation]);
```

Comment:
When moving the hook out of `useProposal` you can also just make it return `verificationStatus` directly instead of having to do a `useMemo` manually here